### PR TITLE
Update Grails 2.x Version

### PR DESCRIPTION
**Pull request by eric-gonzales**
Add support for the following headers:

* Content-Security-Policy
* Public-Key-Pins
* Referrer-Policy
* Strict-Transport-Security
* X-Content-Type-Options
* X-Frame-Options
* X-XSS-Protection

Also upgrade plugin to Grails 2.5.3 and update dependencies.

### CODE REVIEW

File: XframeoptionsGrailsPlugin.groovy [hunk 1]
```diff
@@ -54,14 +54,36 @@ class XframeoptionsGrailsPlugin {
     }
 
     def doWithSpring = {
-        final ConfigUtil configUtil = new ConfigUtil(application)
-        if (configUtil.filterEnabled) {
-            final String headerValue = configUtil.determineHeaderValue()
+        final XFrameOptionsConfiguration xFrameOptionsConfiguration = new XFrameOptionsConfiguration(application)
+        if (xFrameOptionsConfiguration.filterEnabled) {
+            final String xFrameOptionsHeaderValue = xFrameOptionsConfiguration.determineXFrameOptionsHeaderValue()
```
Comment (dbpfindexp) [inline XframeoptionsGrailsPlugin.groovy]: Does it make sense to be able to toggle individual headers somehow?

Comment (eric-gonzales) [reply XframeoptionsGrailsPlugin.groovy]: Yeah, I can fix this up a bit!

Comment (dbpfindexp) [reply XframeoptionsGrailsPlugin.groovy]: I'm cool with it as is just wondering for the fork back to the OG owner's repo would everyone want all these headers in a single groupings or does it make sense to have certain ones and not others?

Comment (eric-gonzales) [reply XframeoptionsGrailsPlugin.groovy]: I just added some configuration options to disable any individual headers not needed.

File: src/groovy/com/mrhaki/grails/plugin/xframeoptions/config/XFrameOptionsConfiguration.groovy [hunk 2]
```diff
@@ -34,10 +38,18 @@ class XFrameOptionsConfiguration {
...
+
         return SCRIPT_SRC_SELF
     }
 
     String determinePublicKeyPinsHeaderValue() {
+        if (config?.publickeypins?.disabled) {
+            return null
+        }
+
         def headerValue = ""
```
Comment (dbpfindexp) [inline src/groovy/com/mrhaki/grails/plugin/xframeoptions/config/XFrameOptionsConfiguration.groovy]: you can just set the initial value to `null` and return this at the end.  In the existing return just set `headerValue`

**Pull request merged**